polkadot/node/core/pvf/src/testing.rs

```diff
@@ -62,6 +62,29 @@ pub fn validate_candidate(
 pub fn get_and_check_worker_paths() -> (PathBuf, PathBuf) {
 	// Only needs to be called once for the current process.
 	static WORKER_PATHS: OnceLock<Mutex<(PathBuf, PathBuf)>> = OnceLock::new();
+
+	fn maybe_build_workers(build_prep: bool, build_exec: bool) {
+		let mut build_args = vec!["build", "--package=polkadot"];
+		if build_prep {
+			build_args.push("--bin=polkadot-prepare-worker");
+		}
+		if build_exec {
+			build_args.push("--bin=polkadot-execute-worker");
+		}
+
+		if build_prep || build_exec {
+			eprintln!("Building workers...");
+
+			let dir = env!("CARGO_MANIFEST_DIR");
+			std::process::Command::new("cargo")
+				.args(build_args)
+				.stdout(std::process::Stdio::piped())
+				.current_dir(dir)
+				.status()
+				.expect("Failed to run the build program");
```

Comment:
We don't check the exit status of the command to see if was a `success()` or not. 

I think it'd be better to quit on failure. For example, if you look at this output with the terminal scrolled to the bottom, it can be easy to miss the errors:

<details>
<pre>
   Compiling polkadot-node-core-pvf-prepare-worker v1.0.0 (/Users/marcinparity/Repos/github.com/paritytech/polkadot-sdk-3/polkadot/node/core/pvf/prepare-worker)
   Compiling polkadot v1.1.0 (/Users/marcinparity/Repos/github.com/paritytech/polkadot-sdk-3/polkadot)
   Compiling polkadot-service v1.0.0 (/Users/marcinparity/Repos/github.com/paritytech/polkadot-sdk-3/polkadot/node/service)
   Compiling polkadot-cli v1.1.0 (/Users/marcinparity/Repos/github.com/paritytech/polkadot-sdk-3/polkadot/cli)
error: no rules expected the token `env`
  --> polkadot/src/bin/prepare-worker.rs:23:2
   |
23 |     env!("SUBSTRATE_CLI_COMMIT_HASH"),
   |     ^^^ no rules expected this token in macro call
   |
   = note: while trying to match sequence start

error[E0601]: `main` function not found in crate `polkadot_prepare_worker`
  --> polkadot/src/bin/prepare-worker.rs:24:3
   |
24 | );
   |   ^ consider adding a `main` function to `polkadot/src/bin/prepare-worker.rs`

For more information about this error, try `rustc --explain E0601`.
error: could not compile `polkadot` (bin "polkadot-prepare-worker") due to 2 previous errors
warning: build failed, waiting for other jobs to finish...
thread 'main' panicked at polkadot/src/bin/prepare-worker.rs:19:1:
trying to run prepare-worker binary with the exit subcommand
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
test worker_common::should_connect ... ok
test prepare_can_run_serially ... ok
Oct 30 16:40:12.350 ERROR runtime: panicked at /Users/marcinparity/Repos/github.com/paritytech/polkadot-sdk-3/polkadot/parachain/test-parachains/adder/src/wasm_validation.rs:36:74:
Executes block: StateMismatch    
test adder::execute_good_block_on_parent ... ok
test adder::execute_bad_block_on_parent ... ok
test adder::execute_can_run_serially ... ok
test adder::execute_good_chain_on_parent ... ok
test adder::stress_spawn ... ok
test worker_common::spawn_immediate_exit ... ok
test worker_common::spawn_timeout ... ok
test terminates_on_timeout ... ok
test deleting_prepared_artifact_does_not_dispute ... ok
test execute_queue_doesnt_stall_if_workers_died ... ok

test result: ok. 12 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 21.01s

   Doc-tests polkadot-node-core-pvf

running 0 tests

test result: ok. 0 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s
</pre>
</details>